src/python/CRABClient/CredentialInteractions.py

```diff
@@ -300,7 +300,13 @@ def createNewMyProxy(self, timeleftthreshold=0, nokey=False):
         myproxytimeleft = 0
         self.logger.debug("Getting myproxy life time left for %s" % self.defaultDelegation["myProxySvr"])
         # return an integer that indicates the number of seconds to the expiration of the proxy in myproxy
-        myproxytimeleft = myproxy.getMyProxyTimeLeft(serverRenewer=True, nokey=nokey)
+        # Also catch the exception in case WMCore encounters a problem with the proxy itself (one such case was #4532)
+        try:
+            myproxytimeleft = myproxy.getMyProxyTimeLeft(serverRenewer=True, nokey=nokey)
+        except Exception as ex:
+            msg = ex._message if hasattr(ex, "_message") else str(ex)
+            raise ProxyCreationException("Problems calculating the time left until the expiration of the proxy."
```

Comment:
should make it clear in the message whether the problem was with myproxy-* or with vom-proxy-*. I'd like to have also the exact command that was issued with its full stdout/err, but I know that this requires a change in WMCore, which could be difficult since there are zilion pieces of code which rely on current signature. But maybe we can consider proposing to add an argument for "report back full log" and when an exception is caught, invoke again the proxy function with "return full log = True")